src/metabase/automagic_dashboards/core.clj

```diff
@@ -278,7 +281,7 @@
   [context dimensions]
   (->> dimensions
        (mapcat (comp (partial make-binding context) first))
-       (group-by (comp :id first :matches val first))
+       (group-by (comp (some-fn :id :name) first :matches val first))
```

Comment:
As a follow on, it would be good to get some schema around this as it's a pretty deep/nested structure that I bet would be pretty sensitive to changes.